bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdataframe.py

```diff
@@ -8,6 +8,164 @@
 
+r'''
+/**
+\class ROOT::RDataFrame
+\brief \parblock \endparblock
+\htmlonly
+<div class="pyrootbox">
+\endhtmlonly
+\anchor python
+## Efficient analysis in Python
+
+You can use RDataFrame in Python thanks to the dynamic Python/C++ translation of PyROOT. In general, the interface
+is the same as for C++, a simple example follows.
+
+~~~{.py}
+df = ROOT.RDataFrame("myTree", "myFile.root")
+sum = df.Filter("x > 10").Sum("y")
+print(sum.GetValue())
+~~~
+
+### User code in the RDataFrame workflow
+
+#### C++ code
+
+In the simple example that was shown above, a C++ expression is passed to the Filter() operation as a string
+(`"x > 0"`), even if we call the method from Python. Indeed, under the hood, the analysis computations run in
+C++, while Python is just the interface language.
+
+To perform more complex operations that don't fit into a simple expression string, you can just-in-time compile
+C++ functions - via the C++ interpreter cling - and use those functions in an expression. See the following
+snippet for an example:
+
+~~~{.py}
+# JIT a C++ function from Python
+ROOT.gInterpreter.Declare("""
+bool myFilter(float x) {
+    return x > 10;
+}
+""")
+
+df = ROOT.RDataFrame("myTree", "myFile.root")
+# Use the function in an RDF operation
+sum = df.Filter("myFilter(x)").Sum("y")
+print(sum.GetValue())
+~~~
+
+To increase the performance even further, you can also pre-compile a C++ library with full code optimizations
+and load the function into the RDataFrame computation as follows.
+
+~~~{.py}
+ROOT.gSystem.Load("path/to/myLibrary.so") # Library with the myFilter function
+ROOT.gInterpreter.Declare('#include "myLibrary.h"') # Header with the definition of the myFilter function
+df = ROOT.RDataFrame("myTree", "myFile.root")
+sum = df.Filter("myFilter(x)").Sum("y")
+print(sum.GetValue())
+~~~
+
+A more thorough explanation of how to use C++ code from Python can be found in the [PyROOT manual](https://root.cern/manual/python/#loading-user-libraries-and-just-in-time-compilation-jitting).
+
+#### Python code
+
+ROOT also offers the option to compile Python functions with fundamental types and arrays thereof using numba.
+Such compiled functions can then be used in a C++ expression provided to RDataFrame.
+
+The function to be compiled is decorated with `ROOT.Numba.Declare`, which allows to specify the parameter and
+return types. See the following snippet for a simple example or the full tutorial [here](pyroot004__NumbaDeclare_8py.html).
+
+~~~{.py}
+@ROOT.Numba.Declare(["float"], "bool")
+def myFilter(x):
+    return x > 10
+
+df = ROOT.RDataFrame("myTree", "myFile.root")
+sum = df.Filter("Numba::myFilter(x)").Sum("y")
+print(sum.GetValue())
+~~~
+
+It also works with collections: `RVec` objects of fundamental types can be transparently converted to/from numpy arrays:
+
+~~~{.py}
+@ROOT.Numba.Declare(['RVec<float>', 'int'], 'RVec<float>')
+def pypowarray(x, y):
+    return x**y
```

Comment:
```suggestion
def pypowarray(numpyvec, pow):
    return numpyvec**pow
```

or alternatively using type annotations